scripts/main/view.js

```diff
@@ -1725,150 +1918,232 @@ view.logs = {
+						":" +
+						String(datetime.getUTCSeconds()).padStart(2, "0") +
+						" UTC"
+					);
+				};
+				const html =
+					logEntries.reduce(function (acc, logEntry) {
+						return (
+							acc +
+							formatDateTime(new Date(logEntry.created_at)) +
+							" -- " +
+							logEntry.type.padEnd(7) +
+							" -- " +
+							logEntry.function +
+							" -- " +
+							logEntry.line +
+							" -- " +
+							logEntry.text +
+							"\n"
+						);
+					}, "<pre>") + "</pre>";
+				$(".logs_diagnostics_view").html(html);
+			};
+
 			view.logs.clearContent();
-			api.post_raw("Logs", {}, function (data) {
-				$(".logs_diagnostics_view").html(data);
-			});
+			api.post("Logs::list", {}, successHandler);
 		},
 	},
 };
 
 view.diagnostics = {
+	/** @returns {void} */
 	init: function () {
 		multiselect.clearSelection();
 
 		view.photo.hide();
-		view.diagnostics.title("Diagnostics");
+		view.diagnostics.title();
 		header.setMode("config");
 		view.diagnostics.content.init();
 	},
 
+	/** @returns {void} */
 	title: function () {
 		lychee.setTitle("Diagnostics", false);
 	},
 
+	/**
+	 * @param {number} update - The update status: `0`: not on master branch;
+	 *                          `1`: up-to-date; `2`: not up-to-date;
+	 *                          `3`: requires migration
+	 * @returns {void}
+	 */
 	clearContent: function (update) {
 		let html = "";
 
 		if (update === 2) {
-			html += view.diagnostics.button("", lychee.locale["UPDATE_AVAILABLE"]);
+			html = view.diagnostics.button("", lychee.locale["UPDATE_AVAILABLE"]);
 		} else if (update === 3) {
-			html += view.diagnostics.button("", lychee.locale["MIGRATION_AVAILABLE"]);
+			html = view.diagnostics.button("", lychee.locale["MIGRATION_AVAILABLE"]);
 		} else if (update > 0) {
-			html += view.diagnostics.button("Check_", lychee.locale["CHECK_FOR_UPDATE"]);
+			html = view.diagnostics.button("Check_", lychee.locale["CHECK_FOR_UPDATE"]);
 		}
 
 		html += '<pre class="logs_diagnostics_view"></pre>';
 		lychee.content.html(html);
 	},
 
+	/**
+	 * @param {string} type
+	 * @param {string} locale
+	 * @returns {string} - HTML
+	 */
 	button: function (type, locale) {
-		let html = "";
-		html += '<div class="clear_logs_update">';
-		html += lychee.html`<a id="${type}Update_Lychee" class="basicModal__button">${locale}</a>`;
-		html += "</div>";
-
-		return html;
+		return `
+			<div class="clear_logs_update">
+				<a id="${type}Update_Lychee" class="basicModal__button">${locale}</a>
+			</div>`;
 	},
 
+	/** @returns {string} */
 	bind: function () {
 		$("#Update_Lychee").on("click", view.diagnostics.call_apply_update);
 		$("#Check_Update_Lychee").on("click", view.diagnostics.call_check_update);
 	},
 
 	content: {
+		/** @returns {void} */
 		init: function () {
-			view.diagnostics.clearContent(false);
-
+			view.diagnostics.clearContent(0);
 			view.diagnostics.content.v_2();
 		},
 
+		/** @returns {void} */
 		v_2: function () {
```

Comment:
This looks like a leftover from Lychee v3->v4 transition. I would merge them into one method...

Reply:
Fixed in:https://github.com/LycheeOrg/Lychee-front/pull/282/commits/058b01318d0d6d7f04fe36b977e64f3f33d63608